# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [9]:
data

id   ad  education  ofjobs  yearsexp  honors  volunteer  military  \
0     b    1          4       2         6       0          0         0   
1     b    1          3       3         6       0          1         1   
2     b    1          4       1         6       0          0         0   
3     b    1          3       4         6       0          1         0   
4     b    1          3       3        22       0          0         0   
5     b    1          4       2         6       1          0         0   
6     b    1          4       2         5       0          1         0   
7     b    1          3       4        21       0          1         0   
8     b    1          4       3         3       0          0         0   
9     b    1          4       2         6       0          1         0   
10    b    1          4       4         8       0          1         0   
11    b    1          4       4         8       0          0         0   
12    b    1          4       4         4       0          0         0   
13    b    1          4       2         4       0          0         0   
14    b    1          4       2         5       0          1         0   
15    b    1          1       3         4       0          0         0   
16    b    1          4       3         5       0          1         0   
17    b    1          4       3         6       0          0         0   
18    b    1          4       2         6       1          1         0   
19    b    1          2       2         8       0          0         0   
20    b    1          4       3         4       0          1         0   
21    b    1          4       3         3       0          0         0   
22    b    1          2       2         2       0          0         0   
23    b    1          4       3         7       0          0         0   
24    b    1          4       3         3       0          1         0   
25    b    1          4       4         6       0          0         0   
26    b    1          1       3         4       0          1         0   
27    b    1          3       4         6       0          1         0   
28    b    1          3       3         7       0          0         0   
29    b    1          4       3         2       0          0         0   
...  ..  ...        ...     ...       ...     ...        ...       ...   
4840  a  93b          4       3         7       0          0         0   
4841  a  93b          4       6         8       0          1         0   
4842  b   95          3       6        16       0          1         0   
4843  b   95          4       4        11       0          0         0   
4844  b   95          3       2         2       0          0         0   
4845  b   95          4       1         5       0          1         0   
4846  a  95b          4       4         2       0          1         1   
4847  a  95b          4       3         7       0          0         0   
4848  a  95b          4       6         8       0          1         0   
4849  a  95b          4       4         6       0          0         0   
4850  b   96          3       5         4       0          0         0   
4851  b   96          2       4        19       0          1         0   
4852  b   96          3       5        13       0          0         0   
4853  b   96          3       7        18       0          1         0   
4854  a  96b          4       6         8       0          1         0   
4855  a  96b          4       3         7       0          0         0   
4856  a  96b          4       4         6       0          0         0   
4857  a  96b          4       4         2       0          1         1   
4858  a  98b          4       4         2       0          1         1   
4859  a  98b          4       4         6       0          0         0   
4860  a  98b          4       6         8       0          1         0   
4861  a  98b          4       3         7       0          0         0   
4862  b  

In [11]:
sum(data[data.race!='b'].call)

235.0

In [8]:
blacks = data[data.race=='b']
blacks

id   ad  education  ofjobs  yearsexp  honors  volunteer  military  \
2       b    1          4       1         6       0          0         0   
3       b    1          3       4         6       0          1         0   
7       b    1          3       4        21       0          1         0   
8       b    1          4       3         3       0          0         0   
9       b    1          4       2         6       0          1         0   
10      b    1          4       4         8       0          1         0   
12      b    1          4       4         4       0          0         0   
14      b    1          4       2         5       0          1         0   
17      b    1          4       3         6       0          0         0   
19      b    1          2       2         8       0          0         0   
20      b    1          4       3         4       0          1         0   
25      b    1          4       4         6       0          0         0   
26      b    1          1       3         4       0          1         0   
28      b    1          3       3         7       0          0         0   
29      b    1          4       3         2       0          0         0   
31      b    1          3       4         6       0          0         0   
33      b    1          4       3         6       1          0         0   
36      b    1          4       3         4       0          0         0   
37      b    1          4       3         3       0          0         0   
38      b    1          3       3         4       0          1         1   
41      b    1          4       3         5       0          1         0   
44      b    1          4       4         5       0          0         0   
45      b    1          3       2         7       0          1         0   
47      b    1          4       3         5       0          1         0   
49      b    1          4       3         5       0          0         0   
50      b    1          4       4         6       0          0         0   
53      b    1          4       2        19       1          0         0   
55      b    1          4       5         7       0          0         1   
57      b    1          2       3         5       0          1         0   
59      b    1          4       3         5       0          0         0   
...   ...  ...        ...     ...       ...     ...        ...       ...   
4808  313    9          4       3         7       0          0         0   
4812  374    9          4       6         8       0          1         0   
4815  262    9          4       5         8       0          0         0   
4818  184    9          4       6         8       0          1         0   
4820  313    9          4       4         2       0          1         1   
4821  229    9          4       4         6       0          0         0   
4823    b   91          3       3         7       0          0         0   
4825    b   91          2       4        16       0          0         0   
4827    a  91b          4       6         8       0          1         0   
4828    a  91b          4       4         6       0          0         0   
4832    b   92          3       5         5       0          0         0   
4833    b   92          4       5        14       0          1         0   
4835    a  92b          4       6         8       0          1         0   
4837    a  92b          4       3         7       0          0         0   
4840    a  93b          4       3         7       0          0         0   
4841    a  93b          4       6         8       0          1         0   
4842    b   95          3       6        16       0          1         0   
4844    b   95          3       2         2       0          0         0   
4848    a  95b          4       6         8       0          1         0   
4849    a  95b          4       4         6       0          0         0   
4850    b   96          3       5         4       0    

In [10]:
whites = data[data.race!='b']
whites

id   ad  education  ofjobs  yearsexp  honors  volunteer  military  \
0       b    1          4       2         6       0          0         0   
1       b    1          3       3         6       0          1         1   
4       b    1          3       3        22       0          0         0   
5       b    1          4       2         6       1          0         0   
6       b    1          4       2         5       0          1         0   
11      b    1          4       4         8       0          0         0   
13      b    1          4       2         4       0          0         0   
15      b    1          1       3         4       0          0         0   
16      b    1          4       3         5       0          1         0   
18      b    1          4       2         6       1          1         0   
21      b    1          4       3         3       0          0         0   
22      b    1          2       2         2       0          0         0   
23      b    1          4       3         7       0          0         0   
24      b    1          4       3         3       0          1         0   
27      b    1          3       4         6       0          1         0   
30      b    1          2       2         6       0          0         0   
32      b    1          3       2         9       0          1         0   
34      b    1          2       2         4       0          0         0   
35      b    1          3       3         5       0          1         0   
39      b    1          4       3         5       0          1         0   
40      b    1          4       2         5       1          0         0   
42      b    1          4       4         7       0          1         0   
43      b    1          4       4         6       0          0         0   
46      b    1          0       2         4       0          1         0   
48      b    1          4       3         4       0          0         0   
51      b    1          4       3         6       0          1         0   
52      b    1          4       5        13       0          0         0   
54      b    1          2       3         5       0          1         0   
56      b    1          4       4        21       0          0         0   
58      b    1          4       4         4       0          0         0   
...   ...  ...        ...     ...       ...     ...        ...       ...   
4811    9    9          4       4         2       0          1         1   
4813    9    9          4       3         7       0          0         0   
4814  313    9          4       6         8       0          1         0   
4816  184    9          4       3         7       0          0         0   
4817  374    9          4       4         2       0          1         1   
4819  401    9          4       4         6       0          0         0   
4822    b   91          4       6         9       0          1         0   
4824    b   91          4       4        11       0          0         0   
4826    a  91b          4       4         2       0          1         1   
4829    a  91b          4       3         7       0          0         0   
4830    b   92          3       3         4       0          0         0   
4831    b   92          3       3         4       0          1         0   
4834    a  92b          4       4         6       0          0         0   
4836    a  92b          4       4         2       0          1         1   
4838    a  93b          4       4         6       0          0         0   
4839    a  93b          4       4         2       0          1         1   
4843    b   95          4       4        11       0          0         0   
4845    b   95          4       1         5       0          1         0   
4846    a  95b          4       4         2       0          1         1   
4847    a  95b          4       3         7       0          0         0   
4851    b   96          2       4        19       0    

In [13]:
whites = whites.call
whites.describe()

count    2435.000000
mean        0.096509
std         0.295346
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max         1.000000
Name: call, dtype: float64

In [14]:
blacks = blacks.call
blacks.describe()

count    2435.000000
mean        0.064476
std         0.245649
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max         1.000000
Name: call, dtype: float64

1. A two-sample test is appropriate for this problem. The central limit theorem (CLT) applies, because we are dealing with two Bernoulli Distributions. As the sample sizes for both populations (black-sounding names and non-black sounding names) increase, the sample means should approach the population means.

2. The null hypothesis is that there is no difference between the rate of callbacks for black-sounding names and non-black sounding names. The alternative hypothesis is there is a difference between the rate of callbacks for the two groups. To be clear, under the alternative hypothesis, the callback rate for blacks could be higher or lower than for non-blacks.

3. 9.7% of white applicants received a callback, versus 6.4% of black applicants. Thus, the difference between the sample means is 3.3 percentage points, or 0.033, with a standard deviation for the distribution of the sample differences of 0.00778 and a margin of error of 0.015. We can reasonably confident that, should this experiment be repeated an infinite number of times, 95% of the differences between the callback rate for blacks and whites would fall between 0.018 and 0.048, or between 1.8% and 4.8%. For the null hypothesis to be true (no difference between blacks and whites) would require a value 2.2 standard deviations from the mean. This corresponds to a p-value of 0.9861. 

4. The observed difference between black and white applicants is statistically significant at the 0.98 significance level. In other words, should the experiment be repeated an infinite number of times, a difference this large or larger would be expected to occur by chance in less than 2% of cases. 

5. Race, or perceived race via the proxy of name, is proven to be an important factor in job application callbacks. Applicants with black-sounding names received 3.3 percentage points fewer callbacks (6.4% versus 9.7%) than applicants without black-sounding names. As a proportion, this means non-blacks received 49.7% more callbacks. The structure of the experiment, in which invented names were randomly assigned to fake resumes, controls for other factors such as the education and experience level of the applicants. The logical next step would be to compare the distributions for other variables and/or construct a linear regression, in which the coefficient for each variable indicates its relative contribution to the sucess of the applicant.